File: raytrace/Object/Albedo/Solid.cpp
#include "Object/Albedo/Solid.hpp"

namespace Object {
	namespace Albedo {
		Solid::Solid(const Object::Color &color)
		: mColor(color)
		{
		}

		Object::Color Solid::color(const Math::Point2D &surfacePoint, const Math::Bivector2D &surfaceProjection) const
		{
			return mColor;
		}
	}
}

File: src/object/albedo/solid.rs
use crate::geo;
use crate::object;

use geo::Bivec2;
use geo::Point2;

use object::Albedo;
use object::Color;

pub struct Solid {
    color: Color,
}

impl Solid {
    pub fn new(color: Color) -> Solid {
        Solid { color }
    }
}

impl Albedo for Solid {
    fn color(&self, _surface_point: Point2, _surface_projection: Bivec2) -> Color {
        return self.color;
    }
}
